client/app/bundles/comments/startup/ServerApp.jsx

```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { Provider } from 'react-redux';
-
-import createStore from '../store/commentsStore';
 import NonRouterCommentsContainer from '../containers/NonRouterCommentsContainer';
+import NavigationBarContainer from '../containers/NavigationBarContainer';
 
-export default props => {
-  const store = createStore(props);
+export default (_props, _railsContext) => {
+  const store = ReactOnRails.getStore('commentsStore');
   return (
     <Provider store={store}>
+      <NavigationBarContainer />
```

Comment:
* https://github.com/reactjs/react-redux/blob/master/src/components/Provider.js takes **children**, so > 1 should be OK
* This file only concerns server rendering, and I can't get tests to pass with only client rendering.

Rob and I spent a couple hours on this and have no idea why this works in manual testing but not in tests.